lib/spack/spack/util/system.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+"""
+Utility methods for interfacing with various system utilities
+"""
+
+import glob
+import os
+import sys
+
+from llnl.util import tty
+from llnl.util.filesystem import join_path
+from llnl.util.lang import memoized
+
+from spack.util.executable import Executable, which
+
+
+@memoized
+def file_command(*args):
+    """Creates entry point to `file` system command with provided arguments"""
+    if sys.platform == "win32":
+        import spack.bootstrap
```

Comment:
You can promote this to a module-level import now (or at least, I think that was the hope for pulling these functions into another module).

Reply:
I tried that, it still causes circular imports due to the Environment module.

Reply:
(just adding circular import error, from perspective of `system.py`)

```
  (system.py)
    import spack.bootstrap
  File "C:\spack_win\spack\lib\spack\spack\bootstrap\__init__.py", line 8, in <module>
    from .core import (
  File "C:\spack_win\spack\lib\spack\spack\bootstrap\core.py", line 51, in <module>
    import spack.user_environment
  File "C:\spack_win\spack\lib\spack\spack\user_environment.py", line 9, in <module>
    import spack.build_environment
  File "C:\spack_win\spack\lib\spack\spack\build_environment.py", line 56, in <module>
    import spack.build_systems.cmake
  File "C:\spack_win\spack\lib\spack\spack\build_systems\cmake.py", line 19, in <module>
    import spack.package_base
  File "C:\spack_win\spack\lib\spack\spack\package_base.py", line 57, in <module>
    from spack.filesystem_view import YamlFilesystemView
  File "C:\spack_win\spack\lib\spack\spack\filesystem_view.py", line 38, in <module>
    import spack.relocate
  File "C:\spack_win\spack\lib\spack\spack\relocate.py", line 28, in <module>
    import spack.util.system as ssys
```